pallets/bridge/src/lib.rs

```diff
@@ -329,285 +375,291 @@ decl_module! {
-		pub fn eval_vote_state(origin, nonce: DepositNonce, src_id: BridgeChainId, prop: Box<<T as Config>::Proposal>) -> DispatchResult {
+		#[pallet::weight((prop.get_dispatch_info().weight + 195_000_000, prop.get_dispatch_info().class, Pays::Yes))]
+		pub fn eval_vote_state(
+			origin: OriginFor<T>,
+			nonce: DepositNonce,
+			src_id: BridgeChainId,
+			prop: Box<<T as Config>::Proposal>,
+		) -> DispatchResult {
 			ensure_signed(origin)?;
 
 			Self::try_resolve_proposal(nonce, src_id, prop)
 		}
 	}
-}
-
-impl<T: Config> Module<T> {
-	// *** Utility methods ***
-
-	/// Checks if who is a relayer
-	pub fn is_relayer(who: &T::AccountId) -> bool {
-		Self::relayers(who)
-	}
 
-	/// Provides an AccountId for the pallet.
-	/// This is used both as an origin check and deposit/withdrawal account.
-	pub fn account_id() -> T::AccountId {
-		MODULE_ID.into_account()
-	}
+	impl<T: Config> Pallet<T> {
+		// *** Utility methods ***
 
-	/// Asserts if a resource is registered
-	pub fn resource_exists(id: ResourceId) -> bool {
-		return Self::resources(id) != None;
-	}
-
-	/// Checks if a chain exists as a whitelisted destination
-	pub fn chain_whitelisted(id: BridgeChainId) -> bool {
-		return Self::chains(id) != None;
-	}
-
-	/// Increments the deposit nonce for the specified chain ID
-	fn bump_nonce(id: BridgeChainId) -> DepositNonce {
-		let nonce = Self::chains(id).unwrap_or_default() + 1;
-		<ChainNonces>::insert(id, nonce);
-		nonce
-	}
+		/// Checks if who is a relayer
+		pub fn is_relayer(who: &T::AccountId) -> bool {
+			Self::relayers(who)
+		}
 
-	// *** Admin methods ***
+		/// Provides an AccountId for the pallet.
+		/// This is used both as an origin check and deposit/withdrawal account.
+		pub fn account_id() -> T::AccountId {
+			MODULE_ID.into_account()
+		}
 
-	/// Set a new voting threshold
-	pub fn set_relayer_threshold(threshold: u32) -> DispatchResult {
-		ensure!(threshold > 0, Error::<T>::InvalidThreshold);
-		<RelayerThreshold>::put(threshold);
-		Self::deposit_event(RawEvent::RelayerThresholdChanged(threshold));
-		Ok(())
-	}
+		/// Asserts if a resource is registered
+		pub fn resource_exists(id: ResourceId) -> bool {
+			return Self::resources(id) != None;
+		}
 
-	/// Register a method for a resource Id, enabling associated transfers
-	pub fn register_resource(id: ResourceId, method: Vec<u8>) -> DispatchResult {
-		<Resources>::insert(id, method);
-		Ok(())
-	}
+		/// Checks if a chain exists as a whitelisted destination
+		pub fn chain_whitelisted(id: BridgeChainId) -> bool {
+			return Self::chains(id) != None;
+		}
 
-	/// Removes a resource ID, disabling associated transfer
-	pub fn unregister_resource(id: ResourceId) -> DispatchResult {
-		<Resources>::remove(id);
-		Ok(())
-	}
+		/// Increments the deposit nonce for the specified chain ID
+		fn bump_nonce(id: BridgeChainId) -> DepositNonce {
+			let nonce = Self::chains(id).unwrap_or_default() + 1;
+			ChainNonces::<T>::insert(id, Some(nonce));
```

Comment:
Didn't understand why in the FRAMEv1 impl it can pass a primitive integer to `insert`. Probably it mapps to `OptionQuery` by default. Switching `OptionQuery` to `ValueQuery` is very dangerous. Could you double check on this?

Reply:
In FRAMEv1 query: ```Option<QueryKind> = Option<OptionQuery>```
In FRAMEv2 query: ```QueryKind = OptionQuery```
I removed Option<> in the value of StorageMap.